src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,293 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
+        IcdManagement::Attributes::ActiveModeDuration::kMetadataEntry,
+        IcdManagement::Attributes::ActiveModeThreshold::kMetadataEntry,
+    };
+    BitFlags<IcdManagement::Feature> featureMap = mICDConfigurationData.GetFeatureMap();
+    const bool hasCIP                           = featureMap.Has(IcdManagement::Feature::kCheckInProtocolSupport);
+    const bool hasUAT                           = featureMap.Has(IcdManagement::Feature::kUserActiveModeTrigger);
+    const bool hasLIT                           = featureMap.Has(IcdManagement::Feature::kLongIdleTimeSupport);
+
+    const AttributeListBuilder::OptionalAttributeEntry optionalEntries[] = {
+        { hasCIP, IcdManagement::Attributes::RegisteredClients::kMetadataEntry },
+        { hasCIP, IcdManagement::Attributes::ICDCounter::kMetadataEntry },
+        { hasCIP, IcdManagement::Attributes::ClientsSupportedPerFabric::kMetadataEntry },
+        { hasUAT, IcdManagement::Attributes::UserActiveModeTriggerHint::kMetadataEntry },
+        { mOptionalAttributeSet.IsSet(IcdManagement::Attributes::UserActiveModeTriggerInstruction::Id),
+          IcdManagement::Attributes::UserActiveModeTriggerInstruction::kMetadataEntry },
+        { hasLIT, IcdManagement::Attributes::OperatingMode::kMetadataEntry },
+        { hasCIP, IcdManagement::Attributes::MaximumCheckInBackOff::kMetadataEntry },
+    };
+    return attributeListBuilder.Append(Span(kMandatoryAttributes), Span(optionalEntries));
 }
 
-CHIP_ERROR ICDManagementAttributeAccess::ReadActiveModeDuration(EndpointId endpoint, AttributeValueEncoder & encoder)
+std::optional<DataModel::ActionReturnStatus> ICDManagementCluster::InvokeCommand(const DataModel::InvokeRequest & request,
+                                                                                 TLV::TLVReader & input_arguments,
+                                                                                 CommandHandler * handler)
 {
-    return encoder.Encode(mICDConfigurationData->GetActiveModeDuration().count());
+    switch (request.path.mCommandId)
+    {
+    case IcdManagement::Commands::StayActiveRequest::Id: {
+// TODO(#32321): Remove #if after issue is resolved
+// Note: We only need this #if statement for platform examples that enable the ICD management server without building the sample
+// as an ICD. Since this is not spec compliant, we should remove this #if statement once we stop compiling the ICD management
+// server in those examples.
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+        Commands::StayActiveRequest::DecodableType commandData;
+        ReturnErrorOnFailure(commandData.Decode(input_arguments));
+
+        IcdManagement::Commands::StayActiveResponse::Type response;
+        response.promisedActiveDuration = Server::GetInstance().GetICDManager().StayActiveRequest(commandData.stayActiveDuration);
+        handler->AddResponse(request.path, response);
+#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
+        break;
```

Comment:
```suggestion
        handler->AddResponse(request.path, response);
        return std::nullopt;
#else
        return Status::Success;  // TODO: WHY? Is this not an error? should we return error instead?
#endif // CHIP_CONFIG_ENABLE_ICD_SERVER
```

Reply:
Yes, this should be error, UnsupportedCommand?